src/main/webapp/js/playGame.js

```diff
@@ -0,0 +1,146 @@
+    gameStage.className = 'center'
+    gameInfo.appendChild(gameStage);
+
+    var theWordHints = document.createElement('h3');
+    theWordHints.innerHTML = 'Hints:';
+    gameInfo.appendChild(theWordHints);
+
+    var starterHint = document.createElement('h3');
+    starterHint.innerHTML = 'Starter: ' + stage.startingHint;
+    gameInfo.appendChild(starterHint);
+
+    var hintsDiv = document.createElement('div');
+    var hintsOl = document.createElement('ol');
+    hintsOl.id = 'hints';
+
+    getHints(stage).forEach(hint => 
+      hintsOl.appendChild(createHintPlaceHolder(hint.hintNumber))
+    );
+
+    hintsDiv.appendChild(hintsOl);
+    gameInfo.appendChild(hintsDiv);
+
+    var enterKeyText = document.createElement('h2');
+    enterKeyText.innerHTML = 'Please Enter The Key To Complete The Stage:';
+    enterKeyText.className = 'center'
+    gameInfo.appendChild(enterKeyText);
+
+    var inputKeyBox = document.createElement('input');
+    inputKeyBox.type = 'text';
+    inputKeyBox.className = 'center';
+    inputKeyBox.classList = 'input-text-color';
+    inputKeyBox.id = 'key-input';
+    inputKeyBox.style = 'width: 30%';
+
+    // This checks if the user clicked enter in the key box
+    inputKeyBox.addEventListener('keydown', function(e) {
+      if (e.which == 13) {
+        checkKey(data, stage.key);
+      }
+    });
+    gameInfo.appendChild(inputKeyBox);
+
+    var buttonToCheckKey = document.createElement('input');
+    buttonToCheckKey.type = 'button';
+    buttonToCheckKey.className = 'center';
+    buttonToCheckKey.value = 'Submit';
+    buttonToCheckKey.addEventListener('click', function() {
+      checkKey(data, stage.key);
+    });
+    gameInfo.appendChild(buttonToCheckKey);
+}
+
+//TODO: Change this so if there is more than one page and the input key is correct,
+// it adds the new stage rather than redirecting to the after game page
+function checkKey(data, key) {
+  var keyInput = document.getElementById('key-input');
+  var inputValue = keyInput.value;
+
+  if (key == inputValue ) {
+      window.location.replace('afterGame.html');
+  } else {
+      window.alert('Wrong key, please try again!');
+  }
+}
+
+function createHintPlaceHolder(hintNum) {
+    var hintLi = document.createElement('li');
+    hintLi.id = hintNum;
+    return hintLi;
+}
+
+async function getStage(stageID) {
+  var currStage;
+  const params = new URLSearchParams();
+  params.append('stageID', stageID)
+  var request = new Request('/load-stage-data', {method: 'POST', body: params});
+  await fetch(request).then(response => response.json()).then((data) => {
+    currStage = data;
+  });
+  return currStage;
+}
+
+function getHints(stage) {
+  return stage.hints;
+}
+
+function addHintMarker(map, latLng, hint, hintNum) {  
+  var infowindow = new google.maps.InfoWindow({
+    content: hint
+  });
+
+  var marker = new google.maps.Marker({
+    position: latLng,
```

Comment:
I feel that position may be less clear. Position could be multiple different things, but I feel that latLng clarifies that this is the latitude and longitude since even from the google maps API what this is representing is "{lat: nextStage.startingLocation.latitude, lng: nextStage.startingLocation.longitude}" which clearly has lat and lng. 

Although the JavaScript doc may clarify this, if someone is looking at it without reading the JavaScript docs, it would be more clear what latLng is over position, unless they don't know what latitude and longitude is.